source/factory.js

```diff
@@ -57,6 +64,9 @@ function createClient(remoteURL, username, password) {
     if (username && username.length > 0) {
         baseOptions.headers.Authorization = authTools.generateBasicAuthHeader(username, password);
     }
+    if (username && username.token_type) {
```

Comment:
Wouldn't it be perhaps better to avoid the `length` check of the first if statement? Perhaps something like:

```javascript
if (username) {
	baseOptions.headers.Authorization = typeof username === "object" ?
		authTools.generateTokenAuthHeader(username) :
		authTools.generateBasicAuthHeader(username, password);
}
```